core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -272,18 +309,24 @@ class CachedDataFrame(df: Dataset[Row], val queryString: String,
       newpls: mutable.ArrayBuffer[ParamLiteral]): Unit = {
     if (allbcplans.nonEmpty && !firstAccess) {
       allbcplans.foreach { case (bchj, refs) =>
+        println(s"Repreparing for bcplan = ${bchj} with new pls = ${newpls.toSet}")
         val broadcastIndex = refs.indexWhere(_.isInstanceOf[Broadcast[_]])
         val newbchj = bchj.transformAllExpressions {
           case pl@ParamLiteral(_, _, p) =>
             val np = newpls.find(_.pos == p).getOrElse(pl)
-            ParamLiteral(np.value, np.dataType, p)
+            val x = ParamLiteral(np.value, np.dataType, p)
+            x.considerUnequal = true
+            x
         }
+        println(s"newbchj = ${newbchj}")
         val tmpCtx = new CodegenContext
         val parameterType = tmpCtx.getClass
         val method = newbchj.getClass.getDeclaredMethod("prepareBroadcast", parameterType)
         method.setAccessible(true)
         val bc = method.invoke(newbchj, tmpCtx)
-        refs(broadcastIndex) = bc.asInstanceOf[(Broadcast[_], String)]._1
+        println(s"replacing bc var = ${refs(broadcastIndex)} with " +
```

Comment:
This entire method I am planning to make either no-op or not call from session.getPlan for now.